src/Hedgehog/Extras/Test/Base.hs

```diff
@@ -300,6 +307,10 @@ leftFail r = GHC.withFrozenCallStack $ case r of
 leftFailM :: (MonadTest m, Show e, HasCallStack) => m (Either e a) -> m a
 leftFailM f = f >>= leftFail
 
+headM :: (MonadTest m, HasCallStack) => [a] -> m a
```

Comment:
Recently I had to rewrite a similar functions in`cardano-node` because it was reused in two places and I could not easily tell which was failing. Perhaps a `Maybe String` would be useful to more easily identify which function is failing. The `Nothing` case could use the empty `String`.

See: https://github.com/input-output-hk/cardano-node/blob/333c701ca9be38e5d955c9d3ee0fab391e85e4f4/cardano-testnet/src/Test/Process.hs#L119

Reply:
This shouldn't be the case because the use of `GHC.withFrozenCallStack` in the function definition cases the failure to indicated the callsite that failed..